### PR TITLE
sci-libs/blis: fix flags/tests issues for 0.9.0, add 1.0

### PR DESCRIPTION
**Pull request by AngryLoki**
This PR closes multiple issues in 0.9.0:
* binary packaging had flaws, because even on x86-64 target machine won't always match host cpu
* crossdev compilation did not work, because correct compilers were not passed
* tests did not work, because `LD_LIBRARY_PATH` had hardcoded `haswell` target
* replaced `make check` with `make testblis-fast` to repeat working CI test set from upstream
* PYTHON_COMPAT: drop 3.9, add 3.12 and 3.13
* 0.9.0 has few checks for aarch64 architecture, but aarch64 was not added to KEYWORDS, as few tests failed

Added 1.0 with the following changes from 0.9:\
* Added patch to not install /usr/include/cblas.h, as it conflicts with sci-libs/lapack
* aarch64 now builds/passes tests (in qemu), can be keyworded later

Note that while cpu_flags_*-based configuration resembles automatic one, it is not fully the same. For example, there are 2 types of Skylake CPUs: with 1 or 2 FMA units. Attempt to enable cpu_flags_x86_avx512vl on CPUs with 1 FMA units will result in reduced performance. Previously this was handled by automatic detection of 1FMA Skylake as `haswell` (and only 2 FMA as `skx`); now user should not specify `cpu_flags_x86_avx512vl` if CPU has degraded AVX-512 support.

Closes: https://bugs.gentoo.org/928018
Closes: https://bugs.gentoo.org/765805
Closes: https://bugs.gentoo.org/929769

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by AndrewAmmerlaan** [comment]
Thanks, a couple of remarks from my side:

>  Added "~arm ~arm64" to KEYWORDS, as it now builds/passes tests in qemu

This should probably go through the arm/arm64 arch teams. We can add the code though, and then file a keyword-request to also gain the KEYWORDS

> Added patch to not install /usr/include/cblas.h, as it conflicts with sci-libs/lapack

Is this the same header file? Or is there a functional difference between both? If so we might want to consider moving instead of removing.

File: sci-libs/blis/blis-1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	CC="$(tc-getCC)" AR="$(tc-getAR)" RANLIB="$(tc-getRANLIB)" ./configure \
+		--enable-verbose-make \
+		--prefix="${BROOT}"/usr \
+		--libdir="${BROOT}"/usr/$(get_libdir) \
+		$(use_enable static-libs static) \
+		--enable-blas \
+		--enable-cblas \
+		"${BLIS_FLAGS[@]}" \
+		--enable-shared \
+		$confname || die
```
Comment (AndrewAmmerlaan) [inline sci-libs/blis/blis-1.0.ebuild:92]:
```suggestion
		$(get_confname) || die
```

We don't really use this anywhere else in this phase right?

File: sci-libs/blis/blis-1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		use arm64 && confname=arm64
+		use cpu_flags_arm_neon && use cpu_flags_arm_v7 && confname=cortexa9
+		use cpu_flags_arm_v8 && confname=cortexa53
+		use cpu_flags_arm_sve && confname=armsve
+	elif use ppc || use ppc64; then
+		confname=power
+		use cpu_flags_ppc_vsx && confname=power7
+		use cpu_flags_ppc_vsx3 && confname=power9
+	fi
+	echo $confname
```
Comment (AndrewAmmerlaan) [inline sci-libs/blis/blis-1.0.ebuild:66]:
```suggestion
	echo ${confname}
```

File: sci-libs/blis/blis-1.0.ebuild (hunk 1 above)
Comment (AngryLoki) [reply sci-libs/blis/blis-1.0.ebuild]: Yes, for now variable is not needed, I'll remove it.

**Comment by AndrewAmmerlaan** [comment]
Thanks :+1:

**Pull request closed without merging**